### PR TITLE
chore: remove librarycommandlet & add tests

### PR DESCRIPTION
**Pull request by AnanyaJha**
### What does this PR do?
This PR removes the libraryCommandlet from the Core package and adds tests to the libraryCommandlet in the utils package

### What issues does this PR fix or reference?
@W-8795013@

### CODE REVIEW

File: packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    this.value += value;
+    this.value += EOL;
+  }
+
+  public clear(): void {}
+  public show(preserveFocus?: boolean | undefined): void;
+  public show(column?: any, preserveFocus?: any): void {}
+  public hide(): void {}
+  public dispose(): void {}
+}
```
Comment (brpowell) [inline packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts:33]: Can we reuse the mock defined in `channelService.test.ts` already? Maybe with a common mocks file.

**Comment by brpowell** [comment]
Nice job getting this to work, I was banging my head against the wall trying to do it last time I tried lol

File: packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts (hunk 1 above)
Comment (AnanyaJha) [reply packages/salesforcedx-utils-vscode/test/unit/commands/commandletExecutor.test.ts]: buena idea, combined the mocks and the stub of vscode

**Comment by AnanyaJha** [comment]
> Nice job getting this to work, I was banging my head against the wall trying to do it last time I tried lol

lmao believe me I was too 🙃

File: packages/salesforcedx-vscode-apex-replay-debugger/src/commands/quickLaunch.ts [hunk 2]
```diff
@@ -127,8 +121,9 @@ export class QuickLaunch {
 }
 
 export class TestDebuggerExecutor extends LibraryCommandletExecutor<string[]> {
-  protected executionName = nls.localize('debug_test_exec_name');
-  protected logName = 'debug_test_replay_debugger';
+  constructor() {
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-apex-replay-debugger/src/commands/quickLaunch.ts:124]: To use the `LibraryCommandletExecutor` from the utils package, we need to provide an output channel - created a new channel for the Replay Debugger extension labeled `Apex Replay Debugger`

**Review by lcampos — APPROVED** [review]

**Pull request merged**